src/core_init_atmosphere/mpas_kd_tree.F

```diff
@@ -187,11 +231,25 @@ recursive subroutine mpas_kd_search_internal(kdtree, query, res, distance)
 
       ! Local Values
       real (kind=RKIND) :: current_distance
+      integer :: tie
 
       current_distance = sum((kdtree % point(:) - query(:))**2)
       if (current_distance < distance) then
          distance = current_distance
          res => kdtree
+      else if (current_distance == distance) then
+          !
+          ! Queries that are equal-distant from two points may end up on inteior cells of
+          ! two decompositions. So ties must be broken arbitrarially so different tasks agree
+          ! on the correct location of of a query.
+          !
+          if (associated(res)) then
+              tie = break_tie(res, kdtree)
+              if (tie == 1) then
```

Comment:
Instead of having an intermediate variable, `tie`, an alternative would be to just test directly on the return value of `break_tie`:
```fortran
if (break_tie(res, kdtree) == 1)
```

Reply:
That sounds good.

Originally I had an `else if` statement that caught when `break_tie` returned 0. Do you think its worth trying to catch when `break_tie` returns 0 and produce an error? i.e. when two points of the KD Tree are the same?

Reply:
I don't think we need to consider it an error if there are two identical points in the kd-tree. If the two points are actually identical, then it doesn't matter which of them we return.

In the longer term, if we decide that it is erroneous to have duplicate points in the kd-tree, then I think the place to check for that would be in the construction of the tree, rather than while searching the kd-tree.

Reply:
That sounds good to me.